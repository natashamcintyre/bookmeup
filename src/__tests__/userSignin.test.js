import React from 'react'
import UserSignin from '../components/userSignin.js'
import { MemoryRouter } from 'react-router-dom'

describe('user sign in form', () => {
  it('renders without crashing', () => {
    const component = mount(
      <MemoryRouter>
        <UserSignin />
      </MemoryRouter>
    )
    expect(component).toMatchSnapshot()
  })

  // const component = mount(
  //   <MemoryRouter>
  //   <UserSignin />
  //   </MemoryRouter>
  // )

  it('has input textbox for username', () => {
    const component = mount(
      <MemoryRouter>
        <UserSignin />
      </MemoryRouter>
    )
    expect(component.exists('input#signin_username')).toBe(true)
  })

  it('has input textbox for password', () => {
    const component = mount(
      <MemoryRouter>
        <UserSignin />
      </MemoryRouter>
    )
    expect(component.exists('input#signin_password')).toBe(true)
  })

  it('has submit button', () => {
    const component = mount(
      <MemoryRouter>
        <UserSignin />
      </MemoryRouter>
    )
    expect(component.exists('button#signin_submit')).toBe(true)
  })

  it('should update state username when text entered', () => {
    const component = shallow(<UserSignin.WrappedComponent />)
    component.find('input#signin_username').simulate('change', {
      target: { value: 'test_username' }
    })
    expect(component.state('username')).toEqual('test_username')
  })

  it('should update state password when text entered', () => {
    const component = shallow(<UserSignin.WrappedComponent />)
    component.find('input#signin_password').simulate('change', {
      target: { value: 'test_password' }
    })
    expect(component.state('password')).toEqual('test_password')
  })

  it('Clear message box on submit', () => {
    const component = mount(<UserSignin.WrappedComponent signinUser={ function (item) { return true }} history={[]} />)
    component.find('input#signin_username').simulate('change', {
      target: { value: 'test_username' }
    })
    component.find('input#signin_password').simulate('change', {
      target: { value: 'test_password' }
    })
    component.find('form#signInForm').simulate('submit')

    expect(component.find('input#signin_username').props().value).toEqual('')
    expect(component.state('username')).toEqual('')
  })
})
