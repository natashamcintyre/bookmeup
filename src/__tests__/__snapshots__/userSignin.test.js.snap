// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user sign in form renders without crashing 1`] = `ReactWrapper {}`;
